conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/DialogueFeignClient.java

```diff
@@ -401,12 +418,29 @@ public String serviceName() {
 
         @Override
         public String endpointName() {
-            return endpoint;
+            return endpointName;
         }
 
         @Override
         public String version() {
             return version;
         }
     }
+
+    @Value.Immutable(builder = false, prehash = true)
+    interface FeignEndpointKey {
```

Comment:
> Otherwise, we'll end up with collisions for overloads that take different args

But there's no consequence to these collisions, right?

The `Endpoint` instances created for these two methods will behave identically, so there's no reason they can't be the same instance. All difference between these endpoints are captured in the request object and applied per request.

Reply:
Talked offline, there is some concern around the per-endpoint concurrency limiters.

Currently, the per-endpoint concurrently limiters have no effect on Feign clients because the `Endpoint` instances are all unique per-request. With the current implement, the concurrency limiters will be too aggressive when there are overloaded methods because overloaded methods will use the same `Endpoint` instance.

To address this we will introduce an additional enrichment header that capture `Method.toString()` and include that in the endpoint cache key.